### PR TITLE
fix: (Core) add fix for Multi-Input defect hunting Sprint 62 bugs

### PR DESCRIPTION
**Pull request by mikerodonnell89**
fixes two multi-input bugs as part of #5420:

- selecting a new item by clicking it anywhere other than the checkbox should close the popover (popover should stay open if checkbox is clicked)
- Tokenizer (and multiinput) - cmd/ctrl+A to select all tokens then you must press delete twice to delete - should only happen once

### CODE REVIEW

**Review by Betrozov — APPROVED** [review]

**Comment by Betrozov** [comment]
@mikerodonnell89 please remove package-lock.json from this PR

File: libs/core/src/lib/multi-input/multi-input.component.ts [hunk 1]
```diff
@@ -419,29 +419,38 @@ export class MultiInputComponent implements
     }
 
     /** @hidden */
-    handleSelect(checked: any, value: any, event?: MouseEvent): void {
+    checkboxChanged(event: boolean, value: any): void {
+        this.handleSelect(event, value, null, true)
+    }
+
+    /** @hidden */
+    handleSelect(checked: any, value: any, event?: MouseEvent, fromCheckbox?: boolean): void {
```
Comment (platon-rov) [inline libs/core/src/lib/multi-input/multi-input.component.ts:427]: `checked` should have boolean type.

**Comment by platon-rov** [comment]
I'm not sure that it should be done in the scope of current PR but I see the component has a lot of `any[]` usage, which can be refactored to use `T[]` to have better type checking.

File: libs/core/src/lib/multi-input/multi-input.component.ts [hunk 2]
```diff
@@ -417,29 +417,38 @@ export class MultiInputComponent implements
...
-        if (this._shouldPopoverBeUpdated(previousLength, this.selected.length)) {
-            this.popoverRef.refreshPosition();
-        }
+            // Handle popover placement update
+            if (this._shouldPopoverBeUpdated(previousLength, this.selected.length)) {
+                this.popoverRef.refreshPosition();
+            }
 
-        this._resetSearchTerm();
+            if (!fromCheckbox) {
```
Comment (Betrozov) [inline libs/core/src/lib/multi-input/multi-input.component.ts:444]: please remove empty `if`

**Review by Betrozov — COMMENTED** [review]
also, please squash all your commits in this PR
and then rename your commit using the new format

**Review by Betrozov — CHANGES_REQUESTED** [review]
also, please squash all your commits in this PR
and then rename your commit using the new format

**Pull request closed without merging**